plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelSpan.java

```diff
@@ -32,9 +34,20 @@ public OTelSpan(String spanName, Span span, org.opensearch.telemetry.tracing.Spa
 
     @Override
     public void endSpan() {
+        if (getAttributeBoolean(TracerContextStorage.SAMPLED) != null && getAttributeBoolean(TracerContextStorage.SAMPLED)) {
+            markParentForSampling();
```

Comment:
AFAIK - This is only needed for Inferred sampling

Reply:
That is true however will remove this once I have the next corresponding evaluation PR in place. If this will be evaluated true then only we will markParentFor Sampling. Will add the comment over it